apps/builder/app/shared/copy-paste/plugin-markdown.ts

```diff
@@ -0,0 +1,222 @@
+import store from "immerhin";
+import { gfm } from "micromark-extension-gfm";
+import type { Root } from "mdast";
+import { fromMarkdown } from "mdast-util-from-markdown";
+import type { Instance, Prop } from "@webstudio-is/project-build";
+import { utils } from "@webstudio-is/project";
+import { nanoid } from "nanoid";
+import {
+  createInstancesIndex,
+  findClosestDroppableTarget,
+  insertInstanceMutable,
+} from "../tree-utils";
+import {
+  instancesIndexStore,
+  propsStore,
+  rootInstanceContainer,
+  selectedInstanceIdStore,
+} from "../nano-states";
+
+const micromarkOptions = { extensions: [gfm()] };
+
+export const mimeType = "text/plain";
+
+// @todo Definition, Strikethrough, Table
+const astTypeComponentMap: Record<string, Instance["component"]> = {
+  paragraph: "Paragraph",
+  heading: "Heading",
+  strong: "Bold",
+  emphasis: "Italic",
+  link: "Link",
+  image: "Image",
+  blockquote: "Blockquote",
+  code: "Code",
+  inlineCode: "InlineCode",
+  list: "List",
+  listItem: "ListItem",
+  thematicBreak: "Separator",
+};
+
+type Options = { generateId?: typeof nanoid };
+
+const createInstance = (
+  {
+    component,
+    node,
+    props,
+  }: {
+    component: Instance["component"];
+    node: Root["children"][number];
+    props: Array<Prop>;
+  },
+  options: Options
+) => {
+  const { generateId = nanoid } = options;
```

Comment:
Why do we have this generateId?

Reply:
Ah, got it. For snapshots. Usually solve this with `.toEqual({id: expect.any(string)})`

Reply:
but you can't use that in snapshots, can you?